File: JavaScript/weather.js
const prompt = require('prompt');
const fetch = require('node-fetch');

prompt.message = '';

const schema = {
	properties: {
		city: {
			message: 'City Name',
		},
	},
};

prompt.get(schema, async function (err, result) {
	const weather = await fetch(
		`https://api.openweathermap.org/data/2.5/weather?q=${result.city}&appid=8ef61edcf1c576d65d836254e11ea420`,
	).then((res) => res.json());
	if (weather.cod != 404) {
		console.log(
			'\33[93m' +
				'\33[1m' +
				`Current weather in ${weather.name}` +
				'\33[0m' +
				'\33[93m' +
				'\33[1m' +
				' is ' +
				'\33[0m' +
				'\33[92m' +
				'\33[1m' +
				weather.weather[0].main +
				'\33[0m' +
				'\33[93m' +
				'\33[1m' +
				' and temperature is ' +
				'\33[0m' +
				'\33[92m' +
				'\33[1m' +
				(weather.main.temp - 273.15).toFixed(2) +
				'°C.' +
				'\33[0m' +
				'\33[93m' +
				'\33[1m' +
				' Humidity is ' +
				'\33[0m' +
				'\33[92m' +
				'\33[1m' +
				weather.main.humidity +
				'%.' +
				'\33[0m',
		);
	} else {
		console.log('City not found.');
	}
});

File: Python/weather.py
import requests

api_key = "8ef61edcf1c576d65d836254e11ea420"
city_name = input("Input City Name")


def requestapi(city_name: str):
    base_url = "https://api.openweathermap.org/data/2.5/weather?"
    complete_url = base_url + "appid=" + api_key + "&q=" + city_name
    response = requests.get(complete_url)
    return response


x = requestapi(city_name).json()
if x["cod"] != "404":
    y = x["main"]
    current_temperature = y["temp"]
    current_humidiy = y["humidity"]
    z = x["weather"]
    weather_description = z[0]["description"]
    print(
        " Temperature in kelvin unit = "
        + str(current_temperature)
        + "\n Humidity (in percentage) = "
        + str(current_humidiy)
        + "\n Description = "
        + str(weather_description)
    )
else:
    print("City Not Found")
